### PR TITLE
Licenses Page

### PR DESCRIPTION
**Pull request by JBreidfjord**
Python script that uses the license-checker npm package to create a JSON file of all licenses, then handles formatting for easier use.
Simple License component that takes data from the formatted JSON and loads onto page.

Script should be run whenever packages are updated, and can be run from either the root sr-velocity folder or from the client folder within.

### CODE REVIEW

File: client/src/pages/licenses/licenses.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        "npx",
+        "license-checker",
+        "--json",
+        "--out",
+        "src/pages/licenses/licenses.json",
+        "--customPath",
+        "src/pages/licenses/licenses_format.json",
+    ]
+)
+
```
[the remark this answers is not in the document]
Comment (JBreidfjord) [reply client/src/pages/licenses/licenses.py:66]: Nice that's a neat trick. The os module made it easy.

File: client/src/pages/licenses/licenses.tsx [hunk 2]
```diff
@@ -1 +1,26 @@
-export {};
+import React from "react";
+
+import "./licenses.css";
+import licenses from "./licenses.json";
+
+const Licenses: React.FC = () => {
+  return (
+    <div>
```
[the remark this answers is not in the document]
Comment (JBreidfjord) [reply client/src/pages/licenses/licenses.tsx]: Changed the class name and added an id to each license and to the containing div. Not the best to have all of the license's div elements share the same id but it shouldn't make a difference for this page anyway.

File: client/src/pages/licenses/licenses.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    # Truncate after "All rights reserved". Lowercase to handle weird case
+    substring = "all rights reserved"
+    if copyright.lower().count(substring) > 0:
+        copyright = copyright[: copyright.lower().index(substring) + len(substring)]
+
+    # Remove site links
+    copyright = " ".join([t for t in copyright.split() if not t.startswith("http")])
+
+    # Replace bracket errors with brackets, then remove all text withing brackets
```
[the remark this answers is not in the document]
Comment (JBreidfjord) [reply client/src/pages/licenses/licenses.py]: Hopefully that makes it a bit more clear! That third pattern is especially ugly with the escapes :poop:

**Comment by JBreidfjord** [comment]
> Can you please a bit more documentation on the python script

Any particular extra documentation you would like to see? I could add docstrings for the functions but I think their names make their purpose pretty understandable already. Other than that it only needs to be run (now from anywhere) so nothing special required to operate!

File: client/src/pages/licenses/licenses.tsx [hunk 4]
```diff
@@ -1 +1,26 @@
...
+import "./licenses.css";
+import licenses from "./licenses.json";
+
+const Licenses: React.FC = () => {
+  return (
+    <div id="licenses">
+      <h1>Licenses</h1>
+      {licenses.map((data, key) => {
+        return (
+          <div key={key} className="license-entry" id="license">
```
Comment (Tijunel) [inline client/src/pages/licenses/licenses.tsx:12]: remove id="license", no need, just style license-entry class name.

**Review by Tijunel — DISMISSED** [review]

**Review by Tijunel — APPROVED** [review]

**Pull request merged**